qp_klp/StandardMetatranscriptomicWorkflow.py

```diff
@@ -74,45 +74,25 @@ def execute_pipeline(self):
         Executes steps of pipeline in proper sequence.
         :return: None
         '''
-        if not self.is_restart:
-            self.pre_check()
+        self.pre_check()
 
-        # this is performed even in the event of a restart.
         self.generate_special_map()
 
-        # even if a job is being skipped, it's being skipped because it was
-        # determined that it already completed successfully. Hence,
-        # increment the status because we are still iterating through them.
-
         self.update_status("Converting data", 1, 9)
-        if "ConvertJob" not in self.skip_steps:
-            # converting raw data to fastq depends heavily on the instrument
-            # used to generate the run_directory. Hence this method is
-            # supplied by the instrument mixin.
-            # NB: convert_raw_to_fastq() now generates fsr on its own
-            results = self.convert_raw_to_fastq()
+
+        self.convert_raw_to_fastq()
 
         self.update_status("Performing quality control", 2, 9)
-        if "NuQCJob" not in self.skip_steps:
-            # NB: quality_control generates its own fsr
-            self.quality_control(self.pipeline)
+
+        self.quality_control()
 
         self.update_status("Generating reports", 3, 9)
-        if "FastQCJob" not in self.skip_steps:
-            # reports are currently implemented by the assay mixin. This is
-            # only because metaranscriptomic runs currently require a failed-
-            # samples report to be generated. This is not done for amplicon
-            # runs since demultiplexing occurs downstream of SPP.
-            results = self.generate_reports()
-            self.fsr_write(results, 'FastQCJob')
+
+        self.generate_reports()
 
         self.update_status("Generating preps", 4, 9)
-        if "GenPrepFileJob" not in self.skip_steps:
-            # preps are currently associated with array mixin, but only
-            # because there are currently some slight differences in how
-            # FastQCJob gets instantiated(). This could get moved into a
-            # shared method, but probably still in Assay.
-            self.generate_prep_file()
+
+        self.generate_prep_file()
```

Comment:
It seems like the stuff in this method is the same as what's in the `StandardMetagenomicWorkflow`, can `execute_pipeline` be pushed to a base class?

Reply:
I could push much of both into the MetaOmics base class off the top of my head. Those two child classes are very similar. Currently they differ only in the string constants we define, off the top of my head. More than happy to include that in my next push.

Reply:
Seems like a good idea?

Reply:
Moved.